src/native/corehost/hostpolicy/deps_resolver.cpp

```diff
@@ -304,6 +303,9 @@ bool deps_resolver_t::probe_deps_entry(const deps_entry_t& entry, const pal::str
             continue;
         }
         pal::string_t probe_dir = config.probe_dir;
+        uint32_t search_options = deps_entry_t::search_options::none;
+        if (m_has_additional_probing_paths || config.only_serviceable_assets)
+            search_options |= deps_entry_t::search_options::file_existence;
```

Comment:
I think we should change the logic here.
In `setup_probe_config` basically if we end up with anything but app and frameworks in the probes - we should enforce file existence checks. This applies to servicing, shared store, additional probing paths... and whatever else we add in the future.